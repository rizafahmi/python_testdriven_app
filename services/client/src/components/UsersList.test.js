import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import UsersList from './UsersList'

const users = [
  {
    active: true,
    email: 'riza@hacktiv8.com',
    id: 1,
    username: 'riza',
    admin: false
  },
  {
    active: true,
    admin: false,
    email: 'rizafahmi@gmail.com',
    id: 2,
    username: 'rizafahmi'
  }
]

test('UsersList renders properly', () => {
  const wrapper = shallow(<UsersList users={users} />)
  expect(wrapper.find('h1').get(0).props.children).toBe('All Users')

  const element = wrapper.find('tbody > tr')
  expect(element.length).toBe(2)
})

test('UsersList renders a snapshot properly', () => {
  const tree = renderer.create(<UsersList users={users} />).toJSON()
  expect(tree).toMatchSnapshot()
})
