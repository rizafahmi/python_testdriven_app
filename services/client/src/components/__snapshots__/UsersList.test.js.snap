// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h1>
    All Users
  </h1>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th>
          User ID
        </th>
        <th>
          Email
        </th>
        <th>
          Username
        </th>
        <th>
          Active
        </th>
        <th>
          Admin
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          riza@hacktiv8.com
        </td>
        <td>
          riza
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          rizafahmi@gmail.com
        </td>
        <td>
          rizafahmi
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
